### PR TITLE
.werft/vm: Add readiness probe

### LINKED ISSUE #7471: Deploy to VM fails on first attempt du to SSH connection errors
### Bug description

`Copying k3s kubeconfig` times out. 
See https://werft.gitpod-dev.com/job/gitpod-build-as-k3s-gitpod.27

### Steps to reproduce

Create a fork of `main`, run via `werft run github -a with-vm=true`

### Workspace affected

_No response_

### Expected behavior

no error. 

### Example repository

gitpod-io/gitpod

### Anything else?

I assume this can not be fixed by increasing the timeout. 
Instead, I assume that the `kubectl port-forward` from https://github.com/gitpod-io/gitpod/blob/9ab23fe0f81f45f7309f53e033fa8eaae60b9d36/.werft/build.ts#L319 is in a bad state and needs to be started later or restarted. 

Workaround: Run the same job again. This will cause the job to connect to the existing VM instead of creating a new one.

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

We fail to establish SSH connection because we try and also timeout before the SSH deamons starts to run.

This PR adds a readiness probe that checks the SSH port. 

## Related Issue(s)

Fixes #7471

## How to test

* Make sure the namespace where de VM is deployed is deleted
* Start a new job
* Keep an eye on the job's log and Harvester UI at the same time
* Notice that we fail the SSH connection before the VM gets ready

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **loujaybee** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @loujaybee` in a comment when ready.

Associated issue: *#7471*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@ArthurSens Oh this is great, such an elegant way to fix it. Hopefully it will be straight forward to check the state of the readiness probe.

**Comment by ArthurSens** [comment]
Alright, we're now using kubectl built-in `wait` command instead of maintaining a timeout logic 🙂

**Comment by ArthurSens** [comment]
@meysholdt, I'm hitting some weird errors (Always after the SSH problem that this PR fixes) regarding gitpod installation. Did you hit them as well when working in the ingress bits?

**Comment by meysholdt** [comment]
These errors? https://gitpod.slack.com/archives/C02PSF5AASV/p1641560514031400
yes, I get them, too :(

File: .werft/vm/manifests.ts
Comment (meysholdt) [inline .werft/vm/manifests.ts:47]: 120 seconds waiting after the VM has started until the readiness probe starts probing seems like a lot. Why do we want an initial delay at all?

Comment (ArthurSens) [reply .werft/vm/manifests.ts:47]: From the tests I've made, not a single VM got ready in less than 3 minutes, and from their [docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)
image

FailureThresholds are the amount of times that we try to open the TCP Socket before setting the pod status to Failed. There is no point trying too early if we know that it just won't be ready

Comment (meysholdt) [reply .werft/vm/manifests.ts:47]: ok, then let's leave it as is.

**Review by meysholdt — APPROVED** [review]
Code changes LGTM and I see that that Werft job reliably executes the changes from this PR :)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 83b80ca032af62abad50e02a79a51550d1c151c0</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7484#pullrequestreview-846659206" title="Approved">meysholdt</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **loujaybee** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @loujaybee` in a comment when ready.

Associated issue: *#7471*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7484#pullrequestreview-846659206" title="Approved">meysholdt</a>*

Associated issue: *#7471*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
